src/DssSpell.t.sol

```diff
@@ -651,4 +650,333 @@ contract DssSpellTest is DssSpellTestBase {
+
+        (, uint256 rate,,,) = vat.ilks("RWA007-A");
+        assertEq(rate, RAY, 'RWA007: bad `rate`'); // rate keeps being 1 RAY
+
+        // wards
+        giveAuth(address(rwaconduitout_007), address(this));
+        // may
+        rwaconduitout_007.mate(address(this));
+        assertEq(rwaconduitout_007.may(address(this)), 1);
+        rwaconduitout_007.hope(address(this));
+        assertEq(rwaconduitout_007.can(address(this)), 1);
+
+        rwaconduitout_007.kiss(address(this));
+        assertEq(rwaconduitout_007.bud(address(this)), 1);
+        rwaconduitout_007.pick(address(this));
+
+        uint256 pushAmount = 100 * WAD; // We push only 100 DAI
+        rwaconduitout_007.push(pushAmount);
+        rwaconduitout_007.quit();
+
+        assertEq(dai.balanceOf(address(rwaconduitout_007)), 0, "RWA007: Output conduit still holds Dai after push()");
+        assertEq(psmGem.balanceOf(address(this)), pushAmount / daiPsmGemDiffDecimals, "RWA007: Psm GEM not sent to destination after push()");
+        assertEq(dai.balanceOf(address(rwaurn_007)), drawAmount - pushAmount, "RWA007: Dai not sent to destination after push()");
+
+        // as we have SF 0 we need to pay exectly the same amount of DAI we have drawn
+        uint256 daiToPay = drawAmount;
+
+        // wards
+        giveAuth(address(rwaconduitinurn_007), address(this));
+        // may
+        rwaconduitinurn_007.mate(address(this));
+        assertEq(rwaconduitinurn_007.may(address(this)), 1);
+
+        // transfer PSM GEM to input conduit
+        psmGem.transfer(address(rwaconduitinurn_007), pushAmount / daiPsmGemDiffDecimals);
+        assertEq(psmGem.balanceOf(address(rwaconduitinurn_007)), pushAmount / daiPsmGemDiffDecimals, "RWA007: Psm GEM not sent to input conduit");
+        
+        // input conduit 'push()' to the urn
+        rwaconduitinurn_007.push();
+
+        assertEq(dai.balanceOf(address(rwaurn_007)), daiToPay, "Balance of the URN doesnt match");
+
+        // repay debt and free our collateral
+        rwaurn_007.wipe(daiToPay);
+        rwaurn_007.free(1 * WAD);
+
+        // check if we get back RWA009 Tokens
+        assertEq(rwagem_007.balanceOf(address(this)), 1 * WAD, "RWA007: gem not sent back to the caller");
+
+        // check if we have 0 collateral and outstanding debt in the VAT
+        (ink, art) = vat.urns("RWA007-A", address(rwaurn_007));
+        assertEq(ink, 0, "RWA007: bad `ink` after free()");
+        assertEq(art, 0, "RWA007: bad `art` after wipe()");
+    }
+
+    function testFailRWA007_DRAW_ABOVE_LINE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 2_000_000 * WAD;
+
+        // setting address(this) as operator
+        hevm.store(address(rwaurn_007), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+
+        // Draw 2mm
+        rwaurn_007.draw(drawAmount);
+    }
+
```

Comment:
```suggestion

    function testFailRWA007_PUSH_ABOVE_LINE() public {
        vote(address(spell));
        scheduleWaitAndCast(address(spell));
        assertTrue(spell.done());

        uint256 drawAmount = 1_000_000 * WAD;

        // setting address(this) as operator
        hevm.store(address(rwaurn_007), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));

        // Draw 1mm
        rwaurn_007.draw(drawAmount);

        // auth
        giveAuth(address(rwaconduitout_007), address(this));

        // pick address(this)
        rwaconduitout_007.hope(address(this)); // allow this to call pick
        rwaconduitout_007.kiss(address(this)); // allow this to be picked
        rwaconduitout_007.pick(address(this));

        // push above line
        uint256 pushAmount = 2_000_000 * WAD;
        rwaconduitout_007.mate(address(this)); // allow this to call push
        rwaconduitout_007.push(pushAmount);    // fail
    }

```

Reply:
Hmm... This one is fully covered by the tests in `mip21-toolkit`

Reply:
I don't see an issue with more tests.